### PR TITLE
Add Access Log Report

### LINKED ISSUE #737: Add endpoints for returning Access Log information
Add GET endpoint that returns `n` of the most recent access log events from the log database. Only site admins have access to this endpoint.

Notes:
  - use new `self.user_is_admin` rather than requiring `root=true` on request
  - possibility of PaginationLite since it's just wrapping a db call?

### PR DESCRIPTION
**Pull request by nagem**
Closes #737 

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

File: api/util.py [hunk 1]
```diff
@@ -84,6 +85,20 @@ def user_perm(permissions, _id, site=None):
...
+    """
+    pattern = re.compile('^[0-9a-zA-Z.@_-]*$')
+    return bool(pattern.match(uid))
+
+def is_group_id(gid):
+    """
+    Checks to make sure uid matches uid regex
+    """
+    pattern = re.compile('^[0-9a-z][0-9a-z.@_-]{0,30}[0-9a-z]$')
+    return bool(pattern.match(gid))
```
Comment (kofalt) [inline api/util.py]: Could these two patterns use the same regex instances as our router?
Maybe move the regexes to this file, and import them into the router?

File: api/handlers/reporthandler.py [hunk 2]
```diff
@@ -92,6 +63,38 @@ def build(self):
...
+
+        Given the output of a mongo aggregation call, checks 'ok' field
+        If not 1.0, 'result' field does not exist or 'result' array is empty,
+        throws APIReportException
+        """
+
+        if output.get('ok') == 1.0:
+            result = output.get('result')
+            if result is not None and len(result) > 0:
+                return result
```
Comment (kofalt) [inline api/handlers/reporthandler.py]: I know this functionality has been duplicated at least once, if you want consider deduping?

File: api/handlers/reporthandler.py [hunk 3]
```diff
@@ -1,84 +1,55 @@
...
-    """
-    Helper function for extracting a singular mongo aggregation result
-
-    If more than one item is in the results array, throws APIReportException
-    """
-
-    results = _get_result_list(output)
-    if len(results) == 1:
-        return results[0]
+class Report(object):
```
Comment (kofalt) [inline api/handlers/reporthandler.py]: :tada:

**Comment by kofalt** [comment]
PR gets an LGTM / shrug, couple optional things you can handle or ignore :slightly_smiling_face:

**Comment by nagem** [comment]
I've got another branch working in here that I added the optional changes to as well as a few other general `Report` improvements, you'll see them later today. Thanks for the review.

**Pull request merged**